packages/docker-node/src/docker-manager.ts

```diff
@@ -0,0 +1,162 @@
+import Docker, { ImageBuildContext } from 'dockerode';
+import fsExtra from 'fs-extra';
+import { IncomingMessage } from 'http';
+import {
+  DockerBadGatewayError,
+  DockerHubConnectionError,
+  DockerNotInstalledError,
+  DockerNotRunningError,
+  DockerServerError,
+  ExecutableNotFoundError,
+  ImageDoesntExistError,
+} from './errors';
+import { Image } from './types';
+
+const DOCKER_SOCKET_PATH = '/var/run/docker.sock';
+
+export class DockerManager {
+  public static async create(): Promise<DockerManager> {
+    if (!(await DockerManager.isInstalled())) {
+      throw new DockerNotInstalledError();
+    }
+
+    if (!(await fsExtra.pathExists(DOCKER_SOCKET_PATH))) {
```

Comment:
[fs.existsSync(path)](https://nodejs.org/api/fs.html#fsexistssyncpath) should work here.